fpga_interchange/xdc.cc

```diff
@@ -106,6 +119,45 @@ static int get_ports(ClientData data, Tcl_Interp *interp, int objc, Tcl_Obj *CON
     }
 }
 
+static int get_cells(ClientData data, Tcl_Interp *interp, int objc, Tcl_Obj *CONST objv[])
+{
+    const Context *ctx = static_cast<const Context *>(data);
+    if (objc == 1) {
+        // Return list of all ports.
+        Tcl_SetStringResult(interp, "Unimplemented");
+        return TCL_ERROR;
+    } else if (objc == 2) {
+        const char *arg0 = Tcl_GetString(objv[1]);
+        IdString cell_name = ctx->id(arg0);
+
+        CellInfo *cell = nullptr;
+        for (auto &cell_pair : ctx->cells) {
```

Comment:
I think you could use `ctx->cells.find` rather than iterating over all the cells?

Reply:
Sure, fixed